internal/datadog/metrics_client.go

```diff
@@ -0,0 +1,99 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package datadog // import "github.com/open-telemetry/opentelemetry-collector-contrib/internal/datadog"
+
+import (
+	"context"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/DataDog/datadog-agent/pkg/trace/metrics"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/metric"
+)
+
+type metricsClient struct {
+	meter  metric.Meter
+	gauges map[string]float64
+	mutex  sync.Mutex
+}
+
+var mutex sync.Mutex
+
+// NewMetricClient returns a new metrics client
+func NewMetricClient(meter metric.Meter) metrics.StatsClient {
+	// We need to lock the mutex here to avoid
+	// duplicate metrics being created by both datadogconnector and exporter
+	mutex.Lock()
+	defer mutex.Unlock()
+	m := &metricsClient{
+		meter:  meter,
+		gauges: make(map[string]float64),
+	}
+	metrics.Client = m
+	return m
+}
```

Comment:
The way things work right now, if an user is using both the Datadog exporter and the Datadog connector on the same Collector, the last one of these to be initialized (which I think would always be the connector right now), would 'win', and its meter would be used. The difference would be on the meter name (`datadogexporter` vs `datadogconnector`) , which affects the instrumentation scope name. I think this scope name can be confusing (the metrics are not *really* from the Datadog exporter or the Datadog connector, but from the trace agent).

My suggestion to fix this is to do something like this:

```suggestion
var initializeOnce sync.Once

// InitializeMetricClient using a meter provider. If the client has already been initialized,
// this function just returns the previous one.
func InitializeMetricClient(mp metric.MeterProvider) {
	initializeOnce.Do(func() {
		m := &metricsClient{
			meter:  mp.Meter("datadog"),
			gauges: make(map[string]float64),
		}
		metrics.Client = m
	})
}
```

That way:
1. Initialization is done only once
2. The meter name (and therefore the scope name of associated metrics) is not specific to the connector or the exporter.